internal/pkg/service/stream/keboolasink/bridge/file.go

```diff
@@ -115,62 +116,67 @@ func (b *Bridge) deleteCredentialsOnFileDelete() {
 	})
 }
 
-func (b *Bridge) registerFileImporter() {
-	b.plugins.RegisterFileImporter(
-		targetProvider,
-		func(ctx context.Context, file *plugin.File) error {
-			// Get authorization token
-			var token *keboolasink.Token
-			err := b.schema.Token().ForSink(file.SinkKey).GetOrNil(b.client).WithResultTo(&token).Do(ctx).Err()
-			if err != nil {
-				return err
-			}
+func (b *Bridge) importFile(ctx context.Context, file *plugin.File, stats statistics.Value) error {
+	start := time.Now()
 
-			// Get file upload credentials
-			var keboolaFile keboolasink.File
-			err = b.schema.File().ForFile(file.FileKey).GetOrErr(b.client).WithResultTo(&keboolaFile).Do(ctx).Err()
-			if err != nil {
-				return err
-			}
+	// Get authorization token
+	token, err := b.schema.Token().ForSink(file.SinkKey).GetOrErr(b.client).Do(ctx).ResultOrErr()
+	if err != nil {
+		return err
+	}
 
-			// Authorized API
-			api := b.publicAPI.WithToken(token.TokenString())
+	// Get file details
+	keboolaFile, err := b.schema.File().ForFile(file.FileKey).GetOrErr(b.client).Do(ctx).ResultOrErr()
+	if err != nil {
+		return err
+	}
 
-			// Check if job already exists
-			var job *keboola.StorageJob
-			if keboolaFile.StorageJobID != nil {
-				job, err = api.GetStorageJobRequest(keboola.StorageJobKey{ID: *keboolaFile.StorageJobID}).Send(ctx)
-				if err != nil {
-					return err
-				}
+	// Authorized API
+	api := b.publicAPI.WithToken(token.TokenString())
 
-				if job.Status == keboola.StorageJobStatusSuccess {
-					return nil
-				}
-			}
+	defer func() {
+		ctx, cancel := context.WithTimeout(ctx, uploadEventSendTimeout)
+		err = b.SendFileImportEvent(ctx, api, time.Since(start), &err, file.FileKey, stats)
+		cancel()
+		if err != nil {
+			b.logger.Warnf(ctx, "unable to send slice upload event: %v", err)
+			return
```

Comment:
This is not supposed to end up with error. Return `nil` and write that we set it to `nil` due to successfull import and we expect to switch the state of file to `Imported` state even though the event was not dispatched.